sunpy/map/mapbase.py

```diff
@@ -2158,19 +2161,28 @@ def draw_quadrangle(self, bottom_left, *, width: u.deg = None, height: u.deg = N
         """
         axes = self._check_axes(axes, allow_non_wcsaxes=False)
 
-        bottom_left, top_right = get_rectangle_coordinates(
-            bottom_left, top_right=top_right, width=width, height=height)
+        if isinstance(bottom_left, u.Quantity):
+            anchor, _, top_right, _ = self._parse_submap_quantity_input(bottom_left, top_right, width, height)
+            width, height = top_right - anchor
+            transform = axes.get_transform(self.wcs)
```

Comment:
As far as I can tell `WCSAxes` does not short circuit this at all, so we probably should.

Reply:
```suggestion
            transform = axes.get_transform(self.wcs if self.wcs is not axes.wcs else 'pixel') 
```